backend/tournesol/views/inconsistencies.py

```diff
@@ -0,0 +1,216 @@
+            criterion = comparison_criterion["criteria"]
+            comparison_score = comparison_criterion["score"]
+
+            try:
+                rating_1 = ratings_map[(entity_1, criterion)]
+                rating_2 = ratings_map[(entity_2, criterion)]
+            except KeyError:
+                continue
+
+            uncertainty = rating_1["uncertainty"] + rating_2["uncertainty"]
+
+            inconsistency = ScoreInconsistencies._calculate_inconsistency(rating_1["score"],
+                                                                          rating_2["score"],
+                                                                          comparison_score,
+                                                                          uncertainty)
+
+            comparisons_analysed_count[criterion] += 1
+            inconsistency_sum[criterion] += inconsistency
+            mean_inconsistency[criterion] = inconsistency_sum[criterion] / \
+                comparisons_analysed_count[criterion]
+
+            if inconsistency >= threshold:
+                inconsistent_comparisons_count[criterion] += 1
+
+                inconsistent_criterion_comparisons.append(
+                    {
+                        "inconsistency": inconsistency,
+                        "criterion": criterion,
+                        "entity_1_uid": entity_1,
+                        "entity_2_uid": entity_2,
+                        "comparison_score": comparison_score,
+                        "entity_1_rating": rating_1["score"],
+                        "entity_2_rating": rating_2["score"],
+                    }
+                )
+
+        inconsistent_criterion_comparisons.sort(key=lambda d: d['inconsistency'], reverse=True)
+
+        response["count"] = len(inconsistent_criterion_comparisons)
+        response["results"] = inconsistent_criterion_comparisons
+        response["stats"] = {}
+        for criterion in criteria_list:
+            response["stats"][criterion] = {
+                "mean_inconsistency": mean_inconsistency[criterion],
+                "inconsistent_comparisons_count": inconsistent_comparisons_count[criterion],
+                "comparisons_count": comparisons_analysed_count[criterion],
+            }
+
+        return response
+
+    @staticmethod
+    def _calculate_inconsistency(entity_1_calculated_rating,
+                                 entity_2_calculated_rating,
+                                 comparison_score,
+                                 uncertainty) -> float:
+        """
+        Calculate the inconsistency between the comparison
+        criterion score and the general rating of the entity.
+
+        Let's note R the rating difference (rating2 - rating1),
+        r its variable counterpart, U the uncertainty and C the
+        comparison (positive if the 2nd entity is preferred).
+        Consider this function i:
+        i(r) = |C - 10r / sqrt(r² + 1)| (cf issue #671 discussion)
```

Comment:
Do you think it may be useful for the users to also include in the API response the value of `10r / sqrt(r² + 1)` and show it to the users as an "expected comparison score"?

Reply:
It would indeed be problematic to tell contributors that their comparison is wrong without indicating an expected score.
But on the other side, it would incentivize them to just set their comparison to the expected value.
Maybe we could, on the frontend, highlight in green on the cursor a range of comparison values that give an acceptable inconsistency (for example, [round(expected score) - 4  , round(expected_score) + 4]).
Another issue is that changing the comparison value will change the rating and thus the expected value. This makes it hard to give an exact range.

in short: I suggest for the backend to indeed return the expected value, and the frontend will convert it to a range of "acceptable" values. It is hard to calculate an exact value for the size of this range, so this can be arbitrary.

Reply:
> I suggest for the backend to indeed return the expected value, and the frontend will convert it to a range of "acceptable" values.

That would be great. I see you have already implemented the change! Thanks